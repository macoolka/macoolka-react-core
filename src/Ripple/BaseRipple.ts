import * as R from 'macoolka-css-components/lib/Ripple';
import {styled} from '../css'

export const RippleRoot = styled(R.RippleRoot);

export const RippleChild = styled(R.RippleChild);
