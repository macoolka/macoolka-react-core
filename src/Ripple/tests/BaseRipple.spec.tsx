import * as React from 'react'
import {RippleRoot,RippleChild} from '../BaseRipple'
import renderer from 'react-test-renderer'
import 'jest-styled-components'


describe('base ripple components',()=>{
    it('RippleBase',()=>{
        const tree = renderer.create(<RippleRoot />).toJSON()
        expect(tree).toMatchSnapshot()
    })
     it('RippleBase visible=true',()=>{
        const tree = renderer.create(<RippleRoot visible={true}/>).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('RippleBase pulsate={true}',()=>{
        const tree = renderer.create(<RippleRoot pulsate={true} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('RippleBase pulsate={true} visible={true}',()=>{
        const tree = renderer.create(<RippleRoot pulsate={true} visible={true}/>).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('RippleChild',()=>{
        const tree = renderer.create(<RippleChild />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('RippleChild pulsate={true}',()=>{
        const tree = renderer.create(<RippleChild pulsate={true}/>).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('RippleChild leaving={true}',()=>{
        const tree = renderer.create(<RippleChild visible={true}/>).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('RippleChild leaving={true} pulsate={true}/',()=>{
        const tree = renderer.create(<RippleChild visible={true} pulsate={true}/>).toJSON()
        expect(tree).toMatchSnapshot()
    }) 
})
