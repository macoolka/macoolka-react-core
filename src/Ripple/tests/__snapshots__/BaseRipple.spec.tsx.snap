// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base ripple components RippleBase 1`] = `
.c1 {
  left: 0px;
  top: 0px;
  width: 50px;
  height: 50px;
  -webkit-animation-duration: 650ms;
  animation-duration: 650ms;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-name: mocoolkarippleEnter;
  animation-name: mocoolkarippleEnter;
  -webkit-animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  opacity: 0.3;
  position: absolute;
  forwarded-component: .c0;
  forwarded-ref: null;
}

<span
  className="c0 c1"
/>
`;

exports[`base ripple components RippleBase pulsate={true} 1`] = `
.c1 {
  left: 0px;
  top: 0px;
  width: 50px;
  height: 50px;
  -webkit-animation-duration: 650ms;
  animation-duration: 650ms;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-name: mocoolkarippleEnter;
  animation-name: mocoolkarippleEnter;
  -webkit-animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  opacity: 0.3;
  position: absolute;
  forwarded-component: .c0;
  forwarded-ref: null;
}

<span
  className="c0 c1"
/>
`;

exports[`base ripple components RippleBase pulsate={true} visible={true} 1`] = `
.c1 {
  left: 0px;
  top: 0px;
  width: 50px;
  height: 50px;
  -webkit-animation-duration: 650ms;
  animation-duration: 650ms;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-name: mocoolkarippleEnter;
  animation-name: mocoolkarippleEnter;
  -webkit-animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  opacity: 0.3;
  position: absolute;
  forwarded-component: .c0;
  forwarded-ref: null;
}

<span
  className="c0 c1"
/>
`;

exports[`base ripple components RippleBase visible=true 1`] = `
.c1 {
  left: 0px;
  top: 0px;
  width: 50px;
  height: 50px;
  -webkit-animation-duration: 650ms;
  animation-duration: 650ms;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-name: mocoolkarippleEnter;
  animation-name: mocoolkarippleEnter;
  -webkit-animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  opacity: 0.3;
  position: absolute;
  forwarded-component: .c0;
  forwarded-ref: null;
}

<span
  className="c0 c1"
/>
`;

exports[`base ripple components RippleChild 1`] = `
.c1 {
  border-radius: 50%;
  left: 0px;
  top: 0px;
  width: 100%;
  height: 100%;
  -webkit-animation-duration: 650ms;
  animation-duration: 650ms;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-name: mocoolkarippleExit;
  animation-name: mocoolkarippleExit;
  -webkit-animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  opacity: 0;
  display: block;
  background-color: currentColor;
  position: absolute;
  forwarded-component: .c0;
  forwarded-ref: null;
}

<span
  className="c0 c1"
/>
`;

exports[`base ripple components RippleChild leaving={true} 1`] = `
.c1 {
  border-radius: 50%;
  left: 0px;
  top: 0px;
  width: 100%;
  height: 100%;
  -webkit-animation-duration: 650ms;
  animation-duration: 650ms;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-name: mocoolkarippleExit;
  animation-name: mocoolkarippleExit;
  -webkit-animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  opacity: 0;
  display: block;
  background-color: currentColor;
  position: absolute;
  forwarded-component: .c0;
  forwarded-ref: null;
}

<span
  className="c0 c1"
/>
`;

exports[`base ripple components RippleChild leaving={true} pulsate={true}/ 1`] = `
.c1 {
  border-radius: 50%;
  left: 0px;
  top: 0px;
  width: 100%;
  height: 100%;
  -webkit-animation-duration: 650ms;
  animation-duration: 650ms;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-name: mocoolkarippleExit;
  animation-name: mocoolkarippleExit;
  -webkit-animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  opacity: 0;
  display: block;
  background-color: currentColor;
  position: absolute;
  forwarded-component: .c0;
  forwarded-ref: null;
}

<span
  className="c0 c1"
/>
`;

exports[`base ripple components RippleChild pulsate={true} 1`] = `
.c1 {
  border-radius: 50%;
  left: 0px;
  top: 0px;
  width: 100%;
  height: 100%;
  -webkit-animation-duration: 650ms;
  animation-duration: 650ms;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-name: mocoolkarippleExit;
  animation-name: mocoolkarippleExit;
  -webkit-animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  animation-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  opacity: 0;
  display: block;
  background-color: currentColor;
  position: absolute;
  forwarded-component: .c0;
  forwarded-ref: null;
}

<span
  className="c0 c1"
/>
`;
